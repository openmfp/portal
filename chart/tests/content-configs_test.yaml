suite: content-configurations
templates:
  - content-configurations/*.yaml
tests:
  - it: matches the snapshot
    values:
      - ../test-values.yaml
    asserts:
      - matchSnapshot: {}
