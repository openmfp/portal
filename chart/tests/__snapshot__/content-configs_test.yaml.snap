matches the snapshot:
  1: |
    apiVersion: core.openmfp.io/v1alpha1
    kind: ContentConfiguration
    metadata:
      name: gardener
      namespace: NAMESPACE
    spec:
      inlineConfiguration:
        content: "\n{\n  \"name\": \"gardener\",\n  \"creationTimestamp\": \"2022-05-17T11:37:17Z\",\n  \"luigiConfigFragment\": [\n    {\n      \"data\": {\n        \"nodes\": [\n          {\n            \"entityType\": \"main.account\",\n            \"pathSegment\": \"gardener\",\n            \"label\": \"Gardener Dashboard\",\n            \"virtualTree\": true,\n            \"initialRoute\": \"namespace\",\n            \"url\": \"https://d.ing.gardener-op.mfp-dev.shoot.canary.k8s-hana.ondemand.com\",\n            \"icon\": \"https://d.ing.gardener-op.mfp-dev.shoot.canary.k8s-hana.ondemand.com/static/assets/logo.svg\",\n            \"loadingIndicator\": {\n              \"enabled\": false\n            },\n            \"requiredIFramePermissions\": {\n              \"sandbox\": [\n                \"allow-forms\", \n                \"allow-modals\",\n                \"allow-popups\", \n                \"allow-popups-to-escape-sandbox\",\n                \"allow-same-origin\", \n                \"allow-scripts\"\n              ],\n              \"allow\": [\n                \"clipboard-read\", \n                \"clipboard-write\"\n              ]\n            }\n          }\n        ]\n      }\n    }\n  ]\n}"
        contentType: json
  2: |
    apiVersion: core.openmfp.io/v1alpha1
    kind: ContentConfiguration
    metadata:
      name: home
      namespace: NAMESPACE
    spec:
      inlineConfiguration:
        content: |2-
          {
            "name": "overview",
            "luigiConfigFragment": [
              {
                "data": {
                  "nodes": [
                    {
                      "entityType": "global",
                      "pathSegment": "home",
                      "hideFromNav": true,
                      "defineEntity": {
                        "id": "main"
                      },
                      "children": [
                        {
                          "pathSegment": "overview",
                          "label": "Overview",
                          "icon": "home",
                          "url": "https://fiddle.luigi-project.io/examples/microfrontends/multipurpose.html",
                          "context": {
                            "title": "Welcome to OpenMFP Portal",
                            "content": " "
                          }
                        }
                      ]
                    }
                  ]
                }
              }
            ]
          }
        contentType: json
